clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -144,13 +159,25 @@ public long metadataExpireMs() {
      */
     public synchronized int requestUpdate() {
         this.needFullUpdate = true;
+        this.backoffUpdateRequests = 0L;
```

Comment:
There are cases where `requestUpdate` is called before the _previous_ update request completed. Wouldn't resetting our backoff count to `0` effectively disable backoff in those cases? Is that the intention?